docs/federated_a_crd.md

```diff
@@ -0,0 +1,399 @@
+
+$ kubectl --validate=false apply -f bar_crd.yaml --context= cluster2
+customresourcedefinition.apiextensions.k8s.io/bars.example.io created
+```
+
+Create "Bar" test resources on memebr clusters and ensure that it's working without any problems.
+
+```
+$ cat bar_test.yaml
+
+apiVersion: example.io/v1
+kind: Bar
+metadata:
+  name: test-crd
+  namespace: test-namespace
+spec:
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      replicas: 3
+
+
+$ kubectl get bars -n test-namspace --context=cluster1
+NAMESPACE        NAME       AGE
+test-namespace   test-crd   14m
+
+
+$ kubectl get bars -n test-namspace --context=cluster2
+NAMESPACE        NAME       AGE
+test-namespace   test-crd   14m
+
+```
+
+### Create the Federation API for your CRD
+
+Now that we've created the CRD in all the clusters we want to federate it to, let's create the federation API for that CRD. The federation API for your CRD is distinct from the CRD itself and is the API surface that declares what the state that should be spread to different clusters is.
+
+There are three pieces of the federation API for a type. <br>The **template type** describes the base definition of a resource that federation should propagate:
+
+```
+$ cat federatedbar_crd.yaml 
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: federatedbars.federation.example.io
+spec:
+  group: federation.example.io
+  names:
+    kind: FederatedBar
+    plural: federatedbars
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            template:
+              type: object
+          type: object
+        status:
+          type: object
+  version: v1alpha1
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+```
+
+The **placement type** holds information about which clusters a federated resource should be spread to:
+
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: federatedbarplacements.federation.example.io
+spec:
+  group: federation.example.io
+  names:
+    kind: FederatedBarPlacement
+    plural: federatedbarplacements
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            clusterNames:
+              items:
+                type: string
+              type: array
+          type: object
+        status:
+          type: object
+  version: v1alpha1
+status:federatedbarsfederatedbars
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+```
+
+The **overrides type** optionally defines how a particular resource should be varied in certain federated clusters:
```

Comment:
How about use `#` to comment out those descriptions? Using those descriptions makes me confused that you have three crd files for template, placement and override, but here you are using one file named as `federatedbar_crd.yaml` for all of the three CRD

Reply:
@gyliu513 Thanks for your feedback. Basically, I made only one crd file where I put entries for all three crds (template, override, placement). The purpose of inserting the textual information between each of these crds is to let the end user understand why they are creating those crds in the first place. If the information is confusing then I can comment out those descriptions as mentioned earlier.

Reply:
Thanks @sohankunkerkar , I prefer that comment this out and make it as a whole YAML file so that people can just copy this template and do some test.

Reply:
@gyliu513 acknowledge.